plugins/modules/ndo_schema_template_bd_dhcp_policy.py

```diff
@@ -0,0 +1,235 @@
+    - The name of the template to change.
+    type: str
+    required: true
+  bd:
+    description:
+    - The name of the BD to manage.
+    type: str
+    required: true
+  dhcp_relay_policy:
+    description:
+    - The name of the DHCP Relay Policy.
+    type: str
+  dhcp_option_policy:
+    description:
+    - The name of the DHCP Option Policy.
+    - When the O(dhcp_option_policy) is provided, the O(dhcp_relay_policy) must also be provided.
+    type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+notes:
+- This module can only be used on versions of MSO that are 4.x or greater.
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Add a new DHCP policy to a BD
+  cisco.mso.ndo_schema_template_bd_dhcp_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    bd: BD1
+    dhcp_relay_policy: ansible_test_relay
+    dhcp_option_policy: ansible_test_option
+    state: present
+
+- name: Query a specific BD DHCP Policy
+  cisco.mso.ndo_schema_template_bd_dhcp_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    bd: BD1
+    dhcp_relay_policy: ansible_test_relay
+    state: query
+  register: query_result
+
+- name: Query all BD DHCP Policies
+  cisco.mso.ndo_schema_template_bd_dhcp_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    bd: BD1
+    state: query
+  register: query_result
+
+- name: Remove a DHCP policy from a BD
+  cisco.mso.ndo_schema_template_bd_dhcp_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    bd: BD1
+    dhcp_relay_policy: ansible_test_relay
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.schema import MSOSchema
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type="str", required=True),
+        template=dict(type="str", required=True),
+        bd=dict(type="str", required=True),
+        dhcp_relay_policy=dict(type="str"),
+        dhcp_option_policy=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["dhcp_relay_policy"]],
+            ["state", "present", ["dhcp_relay_policy"]],
+        ],
```

Comment:
Just a quick suggestion for dhcp_option_policy.

```suggestion
        ],
        required_by={
           ["dhcp_option_policy": "dhcp_relay_policy",
        },
```